Bottom: 125c79b5b8595c4a45f1d0b645c4b8e168a97694
Top:    31a0f3630bf0d8e6044598b3d7427ef423e70329
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-27 15:54:21 -0500

Updates to openldap

Wait on openldap container startup

Occasionally the openldap container would not be ready when attempting
to add the admin and proxyagent accounts. A wait is required to
ensure the service is running properly.

---
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 1ef879c3..c2c9f731 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -142,6 +142,13 @@
 # *****************************************************************************
 # Set the LDAP passwords
 
+- name: wait for LDAP to start
+  wait_for:
+    delay: 5
+    host: '{{ openldap_ip_addr }}'
+    port: 389
+    state: started
+
 - name: set LDAP admin password
   command: 'docker exec -i {{ openldap_container_name }} /docker-entrypoint.sh set_admin_pass {{ openldap_admin_password | quote }}'
 
diff --git a/roles/docker-openldap/tasks/restore.yml b/roles/docker-openldap/tasks/restore.yml
index 370f4797..6c38445f 100644
--- a/roles/docker-openldap/tasks/restore.yml
+++ b/roles/docker-openldap/tasks/restore.yml
@@ -31,14 +31,27 @@
 # restore the openldap backup
 
 - name: stop container (ldap)
-  command: 'docker stop {{ openldap_container_name }}'
+  docker_container:
+    name: '{{ openldap_container_name }}'
+    state: stopped
 
 - name: load ldif files
-  command: 'docker run --rm --volumes-from {{ openldap_dv_name }} -v {{ host_openldap_docker_restore_dir }}:/tmp/import_export {{ openldap_image_name }}:{{ docker_image_tag }} apply_ldif -n {{ item.dbnum }} -l {{ item.ldif_file }}'
+  docker_container:
+    name: '{{ openldap_container_name }}-restore-{{ item.dbnum }}'
+    cleanup: true
+    detach: false
+    volumes_from:
+    - '{{ openssl_dv_name }}'
+    volumes:
+    - '{{ openldap_docker_restore_dir }}:/tmp/import_export:z'
+    image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
+    command: ['apply_ldif', '-n', '{{ item.dbnum }}', '-l', '{{ item.ldif_file }}']
   with_items: '{{ openldap_backup_files }}'
 
 - name: restart container (ldap)
-  command: 'docker start {{ openldap_container_name }}'
+  docker_container:
+    name: '{{ openldap_container_name }}'
+    state: started
 
 # *****************************************************************************
 # cleanup
